### PR TITLE
GameSuccessScreenTest and GameFailureScreenTest

### PR DESCRIPTION
**Pull request by mlytvynchuk**
Added tests for game success and failure screens.

PS. Please review previous PRs before this one. (Based on navigation_updated)

### CODE REVIEW

**Comment by mlytvynchuk** [comment]
Please review previous PRs before this one (Based on navigation_update branch)

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    @Before
+    public void setUp() throws Exception {
+        activityTestRule.launchActivity(MAIN_ACTIVITY_INTENT);
+        LevelManager.init(activityTestRule.getActivity());
+    }
+
+    @Test
+    public void checkIfSuccessTextIsDisplayed() {
+        NavigationHelper.navigateToFragment(activityTestRule.getActivity(), new GameSuccessFragment());
+        onView(withId(R.id.success_text)).check(matches(isDisplayed()));
```
Comment (redls) [inline app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java:56]: nit:add new line

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public void playNextLevel() {
+        LevelManager.setCurrentLevelIndex(FIRST_LEVEL_ID);
+        LevelManager.setLastLevelIndex(FIRST_LEVEL_ID);
+        NavigationHelper.navigateToFragment(activityTestRule.getActivity(), new GameSuccessFragment());
+
+        onView(withId(R.id.next_button)).check(matches(isDisplayed()));
+        onView(withId(R.id.next_button))
+                .perform(click());
+
+        onView(withId(R.id.game_layout)).check(matches(isDisplayed()));
```
Comment (redls) [inline app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java:79]: this test is testing 2 things, maybe split into 2 different tests?

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    @Test
+    public void checkNextButtonIsDisplayedIfNextLevelIsAvailable() {
+        LevelManager.setCurrentLevelIndex(FIRST_LEVEL_ID);
+        LevelManager.setLastLevelIndex(FIRST_LEVEL_ID);
+        NavigationHelper.navigateToFragment(activityTestRule.getActivity(), new GameSuccessFragment());
+
+        if (LevelManager.getLevelsLastIndex() > 0) {
```
Comment (redls) [inline app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java:88]: why do we have this if check?
can we use something like asserThat(LevelManager.getLevelsLastIndex()>0).isTrue or something like that?

File: app/src/main/java/com/google/droidjump/LevelsFragment.java [hunk 4]
```diff
@@ -29,20 +27,22 @@
...
 import com.google.droidjump.models.LevelManager;
+import com.google.droidjump.models.NavigationHelper;
 import java.util.Objects;
 
 /**
  * Displays Levels Screen.
  */
 public class LevelsFragment extends Fragment {
-    private MainActivity activity;
+    private FragmentActivity activity;
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/LevelsFragment.java:39]: why is this change necessary?

File: app/src/main/java/com/google/droidjump/LevelsFragment.java [hunk 5]
```diff
@@ -29,20 +27,22 @@
...
     private LevelsAdapter adapter;
 
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        activity = Objects.requireNonNull((MainActivity) getActivity());
+        activity = getActivity();
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/LevelsFragment.java:45]: maybe cast it here to MainActivity

File: app/src/main/java/com/google/droidjump/LevelsFragment.java (hunk 4 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/LevelsFragment.java]: I don't want depend on class name. In this case you can change activity class in any you want. (In all files we have FragmentActivity). FragmentActivity is a parent class of MainActivity.

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java (hunk 1 above)
Comment (mlytvynchuk) [reply app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java:56]: Fixed.

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java (hunk 2 above)
Comment (mlytvynchuk) [reply app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java:79]: Removed.

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java (hunk 3 above)
Comment (mlytvynchuk) [reply app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java:88]: The user can click on "Next button" and play next level in case if there are any levels (levelsCount > 1). levelsCount = LevelManager.getLevelsLastIndex() + 1. if (LevelManager.getLevelsLastIndex() > 0) we need to show "Next Button", in other cases we do not show the button.

File: app/src/main/java/com/google/droidjump/LevelsFragment.java (hunk 5 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/LevelsFragment.java:41]: Do not think that we need to cast it in case we are using parent class FragmentActivity.

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java (hunk 3 above)
Comment (redls) [reply app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java:86]: Then let's have a test for each case. It is better to avoid these things in tests as bugs may be not detected

File: app/src/main/java/com/google/droidjump/LevelsFragment.java (hunk 4 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/LevelsFragment.java]: Discussed offline and decided to revert change.

File: app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java (hunk 3 above)
Comment (mlytvynchuk) [reply app/src/androidTest/java/com/google/droidjump/GameSuccessScreenTest.java]: Fixed.

**Review by redls — APPROVED** [review]

**Review by redls — APPROVED** [review]
